FarsiLibrary.Win/Controls/FADatePicker.cs

```diff
@@ -477,10 +480,12 @@ protected override void OnMouseDown(MouseEventArgs e)
         {
             base.OnMouseDown(e);
 
-            if (IsReadonly || FormatInfo == FormatInfoTypes.FullDateTime || SelectedDateTime == null || SelectionLength != 0)
-            {
+            if (IsReadonly)
+                return;
+            if (FormatInfo == FormatInfoTypes.FullDateTime || SelectedDateTime == null || SelectionLength != 0)
+                return;
+            if (!mv.MonthViewControl.DefaultCulture.Equals(mv.MonthViewControl.PersianCulture))
```

Comment:
Similar to the above. The shared `IsReadonly` is fine, but the rest that's related to the current culture can be encapsulated in the same component as above.